blog/invictus-logic-apps-testing-framework.md

```diff
@@ -0,0 +1,118 @@
+
+But what exactly do you need to specify in order to really authenticate?  
+Well, below is an overview of the required information:
+
+```csharp
+// The Tenant ID as found in the AAD properties
+string tenantId = "my-tenant-id";
+// The ID of the subscription containing your Logic Apps.
+string subscriptionId = "my-subscription-id";
+// The Client ID of the service principal to be used to authenticate.
+string clientId = "my-client-id";
+// The Client Secret created along with the service principal used for authentication.
+// You might want to look at Arcus.Security.Provides.AzureKeyVault to ensure this value can be stored in Azure Key Vault instead
+string clientSecret = "my-client-secret";
+// The name of the resource group containing the Logic App you want to control
+string resourceGroup = "my-resource-group";
+// The name of the Logic App you want to control.
+string logicAppName = "my-logic-app-name";
+
+// Use the LogicAppAuthentication-class to retrieve the required access token.
+var authentication = LogicAppAuthentication.UsingServicePrincipal(tenantId, subscriptionId, clientId, clientSecret);
+// Once authenticated, use this token to get access to your Logic Apps.
+using (var logicApp = await LogicAppClient.CreateAsync(resourceGroup, logicAppName, authentication))
+{
+}
+```
+
+*Want to read more about how to authenticate and gain access to your Logic Apps?*  
+*Have a look on [this page](https://invictus-integration.github.io/testing-framework/#/logic-apps/authentication).*  
+
+### Controlling an Azure Logic App
+
+One of the first and most important aspects of this framework is the ability to have control over every single Logic App that is part of your interface, only by specifying the name of the resource group and the targeted Logic App. Of course, you also need to authenticate, for which you will need to provide the authentication-token which has been retrieved by the previously mentioned operation.  
+
+But, once the **LogicAppClient**-object has been created, you can perform any of these operations:  
+- Create a Logic App
+- Get the Logic App metadata  
+- Get the Logic App trigger-URL  
+- (Temporarily) enable the Logic App  
+- (Temporarily) update the Logic App definition  
+- (Temporarily) enable static results on specific actions within the Logic App  
+- Trigger a Logic App run  
+- Delete a Logic App  
+
+All of the above operations should provide sufficient possibilities for you to adjust and trigger your Logic Apps to allow for a specific scenario to be tested.
+
+Below is a small example of how you can enable a Logic App during the execution of your test:
+```csharp
+using (var logicApp = await LogicAppClient.CreateAsync(resourceGroup, logicAppName, authentication))
+{
+    // Any action to be performed before enabling the Logic App.
+    await using (await logicApp.TemporaryEnableAsync())
```

Comment:
Shouldn't this be disable?

Reply:
Can go either way - on DEV for instance, you don't want your logic app's to be enabled all the time, so you want to enable it to run your tests and disable it afterwords.

Reply:
Ok, might be confusing to others, so I agree on that.
Regarding the await statement (it's a copy from the docs 🙉) but basicly it's asynchronously going to dispose of the instance performing the action. (IAsyncDisposable vs IDisposable)

Reply:
Aha, scratch the previous bit, we don't have a TemporaryDisableAsync()-method, as we didn't see the need for this.
So, I'll add some context on the TemporaryEnableAsync-operation.

> As you don't want your unit/integration test related Logic App to inflict costs while not being used, here is an example of how you can temporarily enable an Azure Logic App during your test:

Reply:
Thanks for checking!

Tracking in #58